core/input/input.cpp

```diff
@@ -701,18 +701,33 @@ void Input::_parse_input_event_impl(const Ref<InputEvent> &p_event, bool p_is_em
 
 			if (!p_event->is_echo()) {
 				if (p_event->is_action_pressed(E.key)) {
-					action.pressed++;
+					if (jm.is_valid()) {
+						// If axis is already pressed, don't increase the pressed counter.
+						if (!action.axis_pressed) {
+							action.pressed++;
+							action.axis_pressed = true;
+						}
+					} else {
+						action.pressed++;
+					}
+
 					is_pressed = true;
 					if (action.pressed == 1) {
 						action.pressed_physics_frame = Engine::get_singleton()->get_physics_frames();
 						action.pressed_process_frame = Engine::get_singleton()->get_process_frames();
 					}
 				} else {
+					if (jm.is_valid() && !action.axis_pressed) {
+						// Same as above. Don't release axis when not pressed.
+						continue;
```

Comment:
The `continue;` will skip over the line

``` c++
action.exact = InputMap::get_singleton()->event_is_action(p_event, E.key, true);
```
and will skip over the section
```c++
if (is_pressed || action.pressed == 0) {
	action.strength = p_event->get_action_strength(E.key);
	action.raw_strength = p_event->get_action_raw_strength(E.key);
}
```

That means, that for such InputEvents, the attributes `exact`, `strength` and `raw_strength` no longer get updated. This doesn't seem like expected behavior.

Reply:
Not sure about exact, but the second condition will be false (the event is not pressed).

Reply:
`exact` could be solved by moving the assignment above the if-else-clause.

> Action strength for released events is irrelevant.

Before #80859, strength was set set even for released InputEvents, so it was accessible to users. At the moment I don't see any reason for changing this behavior.